src/dmd/mars.d

```diff
@@ -1739,6 +1739,28 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
         }
         else if (arg == "-g") // https://dlang.org/dmd.html#switch-g
             params.symdebug = 1;
+        else if (startsWith(p + 1, "gdwarf=")) // https://dlang.org/dmd.html#switch-gdwarf
+        {
+            static if (TARGET.Linux || TARGET.OSX || TARGET.FreeBSD || TARGET.OpenBSD || TARGET.Solaris || TARGET.DragonFlyBSD)
+            {
+                if (params.dwarf)
+                {
+                    error("`-gdwarf=<version>` can only be provided once!");
+                    break;
+                }
+
+                params.symdebug = 1;
```

Comment:
You might want to use `checkOptions` or the `checkOptionsMixin`. It makes sure the behavior is consistent with other options (e.g. error codes) and handles the common `h|help|?` too, e.g.

```
mixin(checkOptionsMixin("checkGDwarfUsage",
          "`-gdwarf=<versions>` requires a supported DWARF version number (3,4,5)"));
```

(not required though)

Reply:
Done.